idea-plugin/build.gradle

```diff
@@ -0,0 +1,56 @@
+apply plugin: 'java'
+apply plugin: 'com.palantir.external-publish-intellij'
+apply plugin: 'org.jetbrains.grammarkit'
+
+intellij {
+    pluginName = "gradle-consistent-versions"
+    updateSinceUntilBuild = true
+    version = "2024.1"
+    plugins = ['java', 'org.jetbrains.plugins.gradle']
+}
+
+patchPluginXml {
+    pluginDescription = "Language plugin for version.props allowing for auto-lock updating"
+    sinceBuild = '241' // TODO: test against this version of IntelliJ to ensure no regressions
+    untilBuild = ''
+}
+
+// Javadoc fails if there are no public classes to javadoc, so make it stop complaining.
+tasks.named("javadoc", Javadoc) {
+    failOnError = false
+}
+
+sourceSets.main.java.srcDirs 'src/main/generated'
+
+javaVersion {
+    target = 17
+    runtime = 17
+}
+
+dependencies {
+    annotationProcessor 'org.immutables:value'
+    compileOnly 'org.immutables:value::annotations'
+    testImplementation 'org.junit.jupiter:junit-jupiter'
+}
+
+grammarKit {
+    jflexRelease = "1.9.2"
+    grammarKitRelease = "2022.3.2"
+}
+
+generateParser {
+    sourceFile = file('src/main/java/com/palantir/gradle/versions/intellij/VersionProps.bnf')
+    targetRootOutputDir = file('src/main/generated')
+    pathToParser = 'com/palantir/gradle/versions/intellij/parser/VersionPropsParser'
+    pathToPsiRoot = 'com/palantir/gradle/versions/intellij/psi'
+    purgeOldFiles = true
+}
+
+generateLexer {
+    sourceFile = file('src/main/java/com/palantir/gradle/versions/intellij/VersionProps.flex')
+    targetOutputDir = file('src/main/generated/com/palantir/gradle/versions/intellij')
+    targetClass = 'com.palantir.gradle.versions.intellij.VersionPropsLexer'
+    purgeOldFiles = true
+}
+
+compileJava.dependsOn generateParser, generateLexer
```

Comment:
Mmm, this is better now we can generate the code in gradle.

Just a small touchup to this line. It's best to at least prefix the task names with `tasks.`. Otherwise this can match extensions, functions, other things that Gradle magically places in global scope:

```gradle
tasks.compileJava.dependsOn tasks.generateParser, tasks.generateLexer
```

This actually forces the creation of these tasks at configuration time, which can be a performance issue. The absolute "best" way of writing this is:

```gradle
tasks.named('compileJava').configure {
    dependsOn tasks.named('generatedParser'), tasks.named('generateLexer')
}
```

which is as lazy as possible, but is maybe a bit verbose for such a small repo like this.